### ISSUE TITLE
chbench demo fails with docker-compose v2

### ISSUE THREAD

**Issue by ra-kete** [MaterializeInc/materialize#9670]
### What version of Materialize are you using?

The GitHub main branch, though this is not an issue with materialized but with one of the demos.

### What was the issue?

If you run the demo at `demo/chbench` with `mzcompose`, and you have docker-compose version 2 installed, to command fails with:

```
WARN[0000] The "MZ_CHBENCH_SNAPSHOT_DIR" variable is not set. Defaulting to a blank string.
WARN[0000] The "AWS_REGION" variable is not set. Defaulting to a blank string.
WARN[0000] The "AWS_ACCESS_KEY_ID" variable is not set. Defaulting to a blank string.
WARN[0000] The "AWS_SECRET_ACCESS_KEY" variable is not set. Defaulting to a blank string.
WARN[0000] The "AWS_SESSION_TOKEN" variable is not set. Defaulting to a blank string.
1 error(s) decoding:

* error decoding 'Volumes[0]': invalid spec: :/snapshot: empty section between colons
Failed! ERROR: services didn't come up cleanly: materialized
```

The error comes from the `aws-cli` service defined in `demo/chbench/mzcompose.yml`:

```
  aws-cli:
    image: amazon/aws-cli:2.1.18
    environment:
      - AWS_REGION=${AWS_REGION}
      - AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID}
      - AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY}
      - AWS_SESSION_TOKEN=${AWS_SESSION_TOKEN}
    volumes:
      # this is the dir that contains all the individual snapshot directories
      - ${MZ_CHBENCH_SNAPSHOT_DIR}:/snapshot
 ```

`aws-cli` is not a dependency of `demo`, which is why there is no error when docker-compose v1 is used. However, docker-compose v2 always validates the whole compose file, and therefore stumbles over the invalid volume definition, see https://github.com/docker/compose/issues/8841.

The chbench demo is mentioned in the [official documentation](https://materialize.com/docs/demos/business-intelligence/), and there is no mention of the docker-compose v2 incompatibility there, so this seems quite important. I'm not sure if docker-compose v2 is already considered stable at the moment, but it surely will be at some point. At the very least, it is already the default in Arch Linux's package repos.

### Is the issue reproducible? If so, please provide reproduction instructions.

To reproduce, make sure you have:
- docker-compose v2 installed
- the env variable `MZ_CHBENCH_SNAPSHOT_DIR` *not* set

Then run:
```
$ cd demo/chbench
$ ./mzcompose run demo
```

### Workaround

To make the demo work, one can simply set the missing env variable to some dummy value, e.g.:
```
$ env MZ_CHBENCH_SNAPSHOT_DIR=NOTSET ./mzcompose run demo 
```

**Comment by morsapaes** [comment]
Thanks for reporting this issue, @ra-kete! The current demos used in the docs are tangled up in our internal testing and benchmarking suite (which is why you have to use `mzcompose`, for example); this makes them a little less stable and straightforward to run than we'd hope. We're working on making them standalone, so that users don't run into this kind of (unrelated) errors.

(cc @benesch)

**Comment by benesch** [comment]
We're working on making them standalone, so that users don't run into this kind of (unrelated) errors.
> 
> (cc @benesch)

Aw, shoot, one of the original design goals of `mzcompose` was for it to be a first class experience for users running demos. In theory, using mzcompose means we can have CI enforce that our demos still work, even as we make changes to Materialize. That's... of course, not happening here; we've got woefully little automatic testing of most of our demos. And because many of our in-repo demos do double duty as benchmarks, the quality suffers (see e.g. #4559). 

What do you and @ruf-io have planned on the demo front? I'd love to get to a place where we've got a few top-notch demos in the `demo` directory that get automatically tested in CI. If `mzcompose` is causing trouble, there's an escape hatch I envisioned, which was to export the demo directory into its own MaterializeInc/demo repository with every release with the mzcompose.yml file rendered into a normal docker-compose.yml file.

It's late; that might not make much sense; holler if so.

**Comment by benesch** [comment]
To actually get to your issue, @ra-kete, I've started on https://github.com/MaterializeInc/materialize/pull/9690#issuecomment-998191478. There are some irritating incompatibilities with Docker Compose v2 that we'll have to play whack-a-mole with. Once Docker Compose v2.2.3 drops, I'll look into picking some of those issues off.

> ...there is no mention of the docker-compose v2 incompatibility there, so this seems quite important. I'm not sure if docker-compose v2 is already considered stable at the moment, but it surely will be at some point. At the very least, it is already the default in Arch Linux's package repos.

Yeah, we only just learned about Docker Compose v2, which is why it's not mentioned in the docs. There isn't a ton of official clarity on what the plans for v1 are. (Is v1 deprecated? Should everyone switch to v2? The official docs are circumspect.) The fact that Docker for Windows and Docker for Mac just suddenly started shipping the new Docker Compose, but Linux users get to choose their own adventure is... irritating, to say the least. Anyway, I'm sorry you were downstream of this fallout. We'll get it sorted soon!
